lib/web/applications.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package web implements web proxy handler that provides
+// web interface to view and connect to teleport nodes
+package web
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/httplib"
+	"github.com/gravitational/teleport/lib/reversetunnel"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/web/app"
+	"github.com/gravitational/teleport/lib/web/ui"
+
+	"github.com/gravitational/trace"
+	"github.com/julienschmidt/httprouter"
+)
+
+func (h *Handler) siteAppsGet(w http.ResponseWriter, r *http.Request, p httprouter.Params, ctx *SessionContext, site reversetunnel.RemoteSite) (interface{}, error) {
+	clt, err := ctx.GetUserClient(site)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	appServers, err := clt.GetApps(r.Context(), defaults.Namespace)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	proxies, err := clt.GetProxies()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	proxyClusterName, err := h.cfg.ProxyClient.GetClusterName()
+	if err != nil && !trace.IsNotFound(err) {
+		return nil, trace.Wrap(err)
+	}
+
+	proxyName := proxyClusterName.GetClusterName()
+	proxyHost, _, err := services.GuessProxyHostAndVersion(proxies)
+	if err != nil && !trace.IsNotFound(err) {
+		return nil, trace.Wrap(err)
+	}
+	// remove port number if any
+	proxyHost = strings.Split(proxyHost, ":")[0]
```

Comment:
this will fail on IPv6 addresses, use https://golang.org/pkg/net/#SplitHostPort instead